sentry_sdk/scope.py

```diff
@@ -605,11 +605,14 @@ def _drop(cause, ty):
 
         contexts = event.setdefault("contexts", {})
 
-        if has_tracing_enabled(options):
-            if self._span is not None:
-                contexts["trace"] = self._span.get_trace_context()
-        else:
-            contexts["trace"] = self.get_trace_context()
+        if contexts.get("trace") is None:
+            if has_tracing_enabled(options):
+                if self._span is not None:
+                    contexts["trace"] = self._span.get_trace_context()
+                else:
+                    contexts["trace"] = self.get_trace_context()
+            else:
+                contexts["trace"] = self.get_trace_context()
```

Comment:
Could simplify this, should be equivalent
```suggestion
            if has_tracing_enabled(options) and self._span is not None:
                contexts["trace"] = self._span.get_trace_context()
            else:
                contexts["trace"] = self.get_trace_context()
```

Reply:
Also wondering if we need the `has_tracing_enabled(options)` check at all? Wouldn't it be enough to check if there's a `self._span`?